Remora.Discord.Commands/Extensions/CommandTreeExtensions.cs

```diff
@@ -206,30 +206,56 @@ public static Result<IReadOnlyList<IBulkApplicationCommandData>> CreateApplicati
 
             // Translate from options to bulk data
             Optional<ApplicationCommandType> commandType = default;
-            Optional<bool> defaultPermission = default;
+            Optional<bool?> directMessagePermission = default;
+            Optional<IDiscordPermissionSet?> defaultMemberPermissions = default;
             switch (node)
             {
                 case GroupNode groupNode:
                 {
-                    var defaultPermissionAttributes = groupNode.GroupTypes.Select
+                    var memberPermissionAttributes = groupNode.GroupTypes.Select
                         (
-                            t => t.GetCustomAttribute<DiscordDefaultPermissionAttribute>()
+                            t => t.GetCustomAttribute<DiscordDefaultMemberPermissionsAttribute>()
                         )
-                        .ToList();
+                        .ToArray();
 
-                    if (defaultPermissionAttributes.Count > 1)
+                    var directMessagePermisisonAttributes = groupNode.GroupTypes.Select
+                        (
+                            t => t.GetCustomAttribute<DiscordDefaultDMPermissionAttribute>()
+                        )
+                        .ToArray();
+
+                    if (memberPermissionAttributes.Length > 1)
                     {
                         return new UnsupportedFeatureError
```

Comment:
Are you checking this in the tests? See above - remember to test the case where you have two groups with the same name, but only one has the attribute applied (should pass).

Reply:
I'm reusing the pre-existing tests, updated to reflect the current state of the attributes, so, yes?

Reply:
Turns out I already wrote tests for this [here](https://github.com/Nihlus/Remora.Discord/pull/149/files#diff-901547350cc4506bce4ab5d3515ae9db0e1d65b3c15f8ee2172a3b46565cc5e0R241-R257)